lib/StaticNarrative/creator.py

```diff
@@ -49,6 +48,62 @@ def __init__(self: "StaticNarrativeCreator", config: dict[str, Any], token: str)
         ch.setFormatter(formatter)
         self.logger.addHandler(ch)
 
+    def get_narrative_id_from_workspace(self: "StaticNarrativeCreator", ws_ref: str | int) -> str:
+        """Retrieve the Narrative object from a workspace (if one exists).
+
+        :param self: this class
+        :type self: StaticNarrativeCmdLine
+        :param ws_ref: workspace reference (ID or name)
+        :type ws_ref: str | int
+        :raises ValueError: if no Narrative object is found
+        :return: KBaseNarrative.Narrative object UPA
+        :rtype: str
+        """
+        ws_args = "workspaces"
+        if str(ws_ref).isdigit():
+            ws_args = "ids"
+
+        results = self.ws_client.list_objects(
+            {ws_args: [ws_ref], "type": "KBaseNarrative.Narrative"}
+        )
+        if results and results[0] and results[0][0]:
+            return f"{results[0][6]}/{results[0][0]}/{results[0][4]}"
+
+        # no narrative object
+        msg = f"Workspace {ws_ref} did not contain a KBaseNarrative.Narrative object."
+        raise ValueError(msg)
+
+    def create_local_static_narrative(
```

Comment:
this is what the StaticNarrativeCmdLine class did; it's very similar to what the StaticNarrativeCreator does so I combined the classes.